src/middlewared/middlewared/utils/pam.py

```diff
@@ -0,0 +1,302 @@
+            ),
+            pam_module=PAMModule.UNIX
+        ),
+        PAMLine(
+            pam_service=PAMService.ACCOUNT,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 1),
+                PAMControl(PAMResponse.NEW_AUTHTOK_REQD, PAMAction.DONE),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.WINBIND,
+            pam_module_args=('krb5_auth', 'krb5_ccache_type=FILE')
+        )
+    )
+)
+
+SSS_ACCOUNT = PAMConfLines(
+    service=PAMService.ACCOUNT,
+    primary=(
+        PAMLine(
+            pam_service=PAMService.ACCOUNT,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
+                PAMControl(PAMResponse.NEW_AUTHTOK_REQD, PAMAction.DONE),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.UNIX
+        ),
+    ),
+    secondary=(
+        PAMLine(
+            pam_service=PAMService.ACCOUNT,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, PAMAction.OK),
+                PAMControl(PAMResponse.NEW_AUTHTOK_REQD, PAMAction.DONE),
+                PAMControl(PAMResponse.USER_UNKNOWN, PAMAction.IGNORE),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.BAD)
+            ),
+            pam_module=PAMModule.SSS
+        ),
+    )
+)
+
+STANDALONE_AUTH = PAMConfLines(
+    service=PAMService.AUTH,
+    primary=(
+        PAMLine(
+            pam_service=PAMService.AUTH,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 1),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.UNIX
+        ),
+    ),
+)
+
+AD_AUTH = PAMConfLines(
+    service=PAMService.AUTH,
+    primary=(
+        PAMLine(
+            pam_service=PAMService.AUTH,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.UNIX
+        ),
+        PAMLine(
+            pam_service=PAMService.AUTH,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 1),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.WINBIND,
+            pam_module_args=('try_first_pass', 'try_authtok', 'krb5_auth')
+        )
+    )
+)
+
+SSS_AUTH = PAMConfLines(
+    service=PAMService.AUTH,
+    primary=(
+        PAMLine(
+            pam_service=PAMService.AUTH,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.UNIX
+        ),
+        PAMLine(
+            pam_service=PAMService.AUTH,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
```

Comment:
I think this should be 
```
PAMControl(PAMResponse.SUCCESS, 1),
```

Reply:
I agree.  `1` is what's used in current common-auth for a server attached to FreeIPA.